cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala

```diff
@@ -54,44 +57,39 @@ object Service {
-      Success(None)
-    else
-      (annotations.appName, annotations.version) match {
-        case (Some(rawAppName), Some(version)) =>
-          // FIXME there's a bit of code duplicate in Deployment
-          val appName = serviceName(rawAppName)
-          val appVersion = serviceName(s"$appName${Deployment.VersionSeparator}${version.version}")
+               deploymentType: DeploymentType): ValidationNel[String, Option[Service]] =
+    (annotations.appNameValidation |@| annotations.versionValidation) { (rawAppName, version) =>
+      // FIXME there's a bit of code duplicate in Deployment
+      val appName = serviceName(rawAppName)
+      val appVersion = serviceName(s"$appName${Deployment.VersionSeparator}${version.version}")
 
-          val selector =
-            deploymentType match {
-              case CanaryDeploymentType    => "app" -> appName.asJson
-              case RollingDeploymentType   => "app" -> appName.asJson
-              case BlueGreenDeploymentType => "appVersion" -> appVersion.asJson
-            }
-
-          Success(
-            Some(
-              Service(
-                appName,
-                Json(
-                  "apiVersion" -> apiVersion.asJson,
-                  "kind" -> "Service".asJson,
-                  "metadata" -> Json(
-                    "labels" -> Json(
-                      "app" -> appName.asJson),
-                    "name" -> appName.asJson)
-                    .deepmerge(
-                      annotations.namespace.fold(jEmptyObject)(ns => Json("namespace" -> serviceName(ns).asJson))),
-                  "spec" -> Json(
-                    "clusterIP" -> clusterIp.getOrElse("None").asJson,
-                    "ports" -> annotations.endpoints.asJson,
-                    "selector" -> Json(selector))))))
-        case _ =>
-          Failure(new IllegalArgumentException("Unable to generate Service resource: application name is required"))
-      }
+      val selector =
+        deploymentType match {
+          case CanaryDeploymentType    => "app" -> appName.asJson
+          case RollingDeploymentType   => "app" -> appName.asJson
+          case BlueGreenDeploymentType => "appVersion" -> appVersion.asJson
+        }
 
+      if (annotations.endpoints.isEmpty)
+        None
+      else
+        Some(
+          Service(
+            appName,
+            Json(
+              "apiVersion" -> apiVersion.asJson,
+              "kind" -> "Service".asJson,
+              "metadata" -> Json(
+                "labels" -> Json(
+                  "app" -> appName.asJson),
+                "name" -> appName.asJson)
+                .deepmerge(
+                  annotations.namespace.fold(jEmptyObject)(ns => Json("namespace" -> serviceName(ns).asJson))),
+              "spec" -> Json(
+                "clusterIP" -> clusterIp.getOrElse("None").asJson,
+                "ports" -> annotations.endpoints.asJson,
+                "selector" -> Json(selector)))))
```

Comment:
If we do need the `appName` in the `BlueGreenDeploymentType`, this needs to be modified:

```
 "selector" -> selector
```